rinja_derive/src/generator.rs

```diff
@@ -1657,18 +1657,11 @@ impl<'a> Generator<'a> {
         buf: &mut Buffer,
         name: &str,
         args: &[WithSpan<'_, Expr<'_>>],
-        node: &WithSpan<'_, T>,
+        _node: &WithSpan<'_, T>,
     ) -> Result<DisplayWrap, CompileError> {
-        if cfg!(not(feature = "humansize")) {
-            return Err(ctx.generate_error(
-                &format!("the `{name}` filter requires the `humansize` feature to be enabled"),
-                node,
-            ));
-        }
-
         // All filters return numbers, and any default formatted number is HTML safe.
         buf.write(format_args!(
-            "rinja::filters::HtmlSafeOutput(rinja::filters::{name}(",
+            "rinja::filters::HtmlSafeOutput(rinja::filters::{name}("
         ));
         self._visit_args(ctx, buf, args)?;
         buf.write(")?)");
```

Comment:
```suggestion
        buf.write(format_args!(
            "rinja::filters::HtmlSafeOutput(\
                rinja::filters::{name}(\
                    rinja::helpers::get_primitive_value(&("
        ));
        self._visit_args(ctx, buf, args)?;
        buf.write(
            "\
                    )) as rinja::helpers::core::primitive::f32\
                )?\
            )"
        );
```

So `filesizeformat()` does only need to accept f32.

Reply:
I'm a bit shared about this. With the current approach, if users have something like a newtype, they can just implement the `ToF32` trait on it to make it work. With your suggestion, they'd need to make a different kind of conversion before being able to call the filter. It sounds a bit less practical.

Reply:
I would rather make `rinja::helpers::PrimitiveType` public, so a user could implement something like that
```rust
impl PrimitiveType for DiskSpaceRequirements {
    type Value = u64;

    #[inline]
    fn get(&self) -> u64 {
        self.unpacked_size
    }
}
```

Otherwise, if you want to add an explicit trait, I would rename it to something like `AsFilesize`.

Reply:
Good idea! Let's go with `PrimitiveType` as public trait. :)